### PR TITLE
feat(cli): alert profiles command

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Signed-off-by: Darren Murray <darren.murray@lacework.net>

## Summary

Add alert profile commands to cli.

`lacework alert-profile list`
`lacework alert-profile show <id>`
`lacework alert-profile delete <id>`
`lacework alert-profile create`

## How did you test this change?
Manually run commands
Add unit test `cmd/alert_profile_test.go`

## Issue 
https://lacework.atlassian.net/browse/ALLY-962

### CODE REVIEW

File: cli/cmd/alert_profiles.go [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+	"github.com/lacework/go-sdk/api"
+)
+
+var (
+	// alert-profiles command is used to manage lacework alert profiles
+	alertProfilesCommand = &cobra.Command{
+		Use:     "alert-profile",
+		Aliases: []string{"alert-profiles", "ap"},
+		Short:   "Manage alert profiles",
+		Long: `Manage alert profiles to define how your LQL queries get consumed into alerts
```
Comment (afiune) [inline cli/cmd/alert_profiles.go:39]:
```suggestion
		Long: `Manage alert profiles to define how your LQL queries get consumed into alerts.
```

File: cli/cmd/alert_profiles.go [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+An alert profile consists of the name of the new profile, the name of an existing profile from which
+the new profile extends and a list of alert templates`,
```
Comment (afiune) [inline cli/cmd/alert_profiles.go:42]:
```suggestion
the new profile extends and a list of alert templates.`,
```

File: cli/cmd/alert_profiles.go [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+				cli.OutputHuman(fmt.Sprintf(msg, cli.Account))
+				return nil
+			}
+			if cli.JSONOutput() {
+				return cli.OutputJSON(alertProfiles)
+			}
+
+			var rows [][]string
+			for _, profile := range alertProfiles.Data {
+				rows = append(rows, []string{profile.Guid, profile.Extends})
```
Comment (afiune) [inline cli/cmd/alert_profiles.go:79]: can we order this list

File: cli/cmd/alert_profiles_test.go [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    FILE_PERMISSIONS       
+    LINK_ABS_DEST_PATH     
+    _POLICY_TITLE          
+    _POLICY_SEVERITY       
+    FILE_CREATED_TIME      
+    FILE_MODIFIED_TIME     
+    FILEDATA_HASH          
+                           
+
+`
```
Comment (afiune) [inline cli/cmd/alert_profiles_test.go:206]: What about this output instead?
```
        ALERT PROFILE ID                 EXTENDS
---------------------------------+---------------------
  LW_HE_MACHINES_DEFAULT_PROFILE   LW_LPP_BaseProfile

                                                          ALERT TEMPLATES
-----------------------------------------------------------------------------------------------------------------------------------
              NAME                       EVENT NAME                      DESCRIPTION                        SUBJECT
  ---------------------------+--------------------------------+--------------------------------+---------------------------------
    HE_Machine_NewViolation    LW Host Entity Machine           {{_OCCURRENCE}} Violation for    {{_OCCURRENCE}} violation
                               Violation Alert                  machine {{HOSTNAME}}             detected for machine
                                                                                                 {{HOSTNAME}}
    HE_Machine_PolicyChanged   LW Host Entity Machine           {{_OCCURRENCE}} Violation for    {{_OCCURRENCE}} violation
                               Violation Alert                  machine {{HOSTNAME}}             detected for machine
                                                                                                 {{HOSTNAME}}
    HE_Machine_Violation       LW Host Entity Machine           {{_OCCURRENCE}} Violation for    {{_OCCURRENCE}} violation
                               Violation Alert                  machine {{HOSTNAME}}             detected for machine
                                                                                                 {{HOSTNAME}}

                                                                       FIELDS
--------------------------------------------------------------------------------------------------------------------------------------
  OS, MID, RECORD_CREATED_TIME ,_EVENT_COUNT ,_POLICY_ID ,_SEVERITY ,KERNEL_ARGS ,HOSTNAME ,DEFAULT_ROUTER ,OS_VERSION ,
  LAST_BOOT_TIME ,KERNEL_VERSION ,KERNEL_RELEASE ,TAGS ,_PRIMARY_TAG ,ROUTE ,OS_DESC ,DOMAIN ,_POLICY_DESCRIPTION ,LAST_BOOT_REASON ,
  MEMORY_INFO ,_RISK ,_OCCURRENCE ,CPU_INFO ,KERNEL ,_POLICY_TITLE ,MACHINE_ID ,_POLICY_SEVERITY

To use fields inside the description or subject of a template enclose the name between double brakets. For example: '{{FIELD_NAME}}'
```

File: cli/cmd/alert_profiles.go [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+			if !cli.InteractiveMode() {
+				return errors.New("interactive mode is disabled")
+			}
+
+			response, err := promptCreateAlertProfile()
+			if err != nil {
+				return errors.Wrap(err, "unable to create alert profile")
+			}
+
+			cli.OutputHuman(fmt.Sprintf("The alert profile was created with GUID %s \n", response.Data.Guid))
```
Comment (afiune) [inline cli/cmd/alert_profiles.go:145]: We display GUID here but in the SHOW table we display ID. Can we match them?

File: cli/cmd/alert_profiles.go [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+
+	return api.AlertTemplate{
+		Name:        answers.Name,
+		EventName:   answers.EventName,
+		Description: answers.Description,
+		Subject:     answers.Subject,
+	}
+}
+
+func getAlertProfiles() []string {
```
Comment (afiune) [inline cli/cmd/alert_profiles.go:321]: can we unit test this function?

File: cli/cmd/alert_profiles.go [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+	if err != nil {
+		return nil
+	}
+	var profiles = make([]string, 0)
+	for _, p := range response.Data {
+		// profiles can only extend from 'LW_' profiles with >= 1 alert template
+		if strings.HasPrefix(p.Guid, "LW_") && len(p.Alerts) >= 1 {
+			profiles = append(profiles, p.Guid)
+		}
+	}
```
Comment (afiune) [inline cli/cmd/alert_profiles.go:333]: can we order this list?

File: cli/cmd/alert_profiles.go [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+			headers = append(headers, []string{alertProfile.Guid, alertProfile.Extends})
+			cli.OutputHuman(renderSimpleTable([]string{"ID", "EXTENDS"}, headers))
+			cli.OutputHuman("\n")
+			cli.OutputHuman(buildAlertProfileDetailsTable(alertProfile))
+
+			return nil
+		},
+	}
+
+	// delete command is used to remove a lacework alert profile by resource id
```
Comment (afiune) [inline cli/cmd/alert_profiles.go:115]:
```suggestion
	// delete command is used to remove a lacework alert profile by id
```

File: cli/cmd/alert_profiles.go [hunk 9]
```diff
@@ -0,0 +1,336 @@
...
+				return cli.OutputJSON(alertProfiles)
+			}
+
+			var rows [][]string
+			for _, profile := range alertProfiles.Data {
+				rows = append(rows, []string{profile.Guid, profile.Extends})
+			}
+
+			cli.OutputHuman(renderSimpleTable([]string{"ID", "EXTENDS"}, rows))
+			return nil
```
Comment (afiune) [inline cli/cmd/alert_profiles.go:83]: Are these IDs or GUIDs?

File: cli/cmd/alert_profiles.go [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+		Use:   "delete <alert_profile_id>",
+		Short: "Delete a alert profile",
+		Long:  "Delete a single alert profile by it's ID.",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			err := cli.LwApi.V2.Alert.Profiles.Delete(args[0])
+			if err != nil {
+				return errors.Wrap(err, "unable to delete alert profile")
+			}
+			cli.OutputHuman(fmt.Sprintf("The alert profile with GUID %s was deleted \n", args[0]))
```
Comment (afiune) [inline cli/cmd/alert_profiles.go:126]:
```suggestion
			cli.OutputHuman("The alert profile with GUID %s was deleted \n", args[0])
```

File: integration/test_resources/help/alert-profile [hunk 11]
```diff
@@ -0,0 +1,35 @@
+Manage alert profiles to define how your LQL queries get consumed into alerts
+
+An alert profile consists of the name of the new profile, the name of an existing profile from which
+the new profile extends and a list of alert templates
```
Comment (benradstone) [inline integration/test_resources/help/alert-profile:4]:
```suggestion
the new profile extends and a list of alert templates.
```

File: integration/test_resources/help/alert-profile [hunk 12]
```diff
@@ -0,0 +1,35 @@
+Manage alert profiles to define how your LQL queries get consumed into alerts
```
Comment (benradstone) [inline integration/test_resources/help/alert-profile:1]:
```suggestion
Manage alert profiles to define how your LQL queries get consumed into alerts.
```

File: integration/test_resources/help/alert-profile_delete [hunk 13]
```diff
@@ -0,0 +1,21 @@
+Delete a single alert profile by it's ID.
```
Comment (benradstone) [inline integration/test_resources/help/alert-profile_delete:1]:
```suggestion
Delete a single alert profile by its ID.
```

File: integration/test_resources/help/alert-profile_show [hunk 14]
```diff
@@ -0,0 +1,24 @@
+Show a single alert profile by it's ID.
```
Comment (benradstone) [inline integration/test_resources/help/alert-profile_show:1]:
```suggestion
Show a single alert profile by its ID.
```

**Review by benradstone — CHANGES_REQUESTED** [review]
Some spelling and grammatical comments, feel free to commit them.

File: cli/cmd/alert_profiles.go (hunk 9 above)
Comment (dmurray-lacework) [reply cli/cmd/alert_profiles.go:83]: IDs

**Review by benradstone — APPROVED** [review]

**Pull request merged**